src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -0,0 +1,269 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration.Suppression;
+using SonarQube.Client.Models;
+using SonarQube.Client.Services;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class SonarQubeIssueProviderTests
+    {
+        private Mock<ISonarQubeService> mockSqService;
+        private Mock<IActiveSolutionBoundTracker> mockTracker;
+        private Mock<ITimerFactory> mockTimerFactory;
+        private Mock<ITimer> mockTimer;
+
+        /// <summary>
+        /// Wired up to the Timer.Start/Stop mock methods to track the current timer status
+        /// </summary>
+        private bool timerRunning;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            mockSqService = new Mock<ISonarQubeService>();
+            mockTracker = new Mock<IActiveSolutionBoundTracker>();
+
+            mockTimerFactory = new Mock<ITimerFactory>();
+            mockTimer = new Mock<ITimer>();
+
+            mockTimerFactory.Setup(x => x.Create())
+                .Returns(mockTimer.Object)
+                .Verifiable();
+
+            mockTimer.SetupSet(x => x.Interval = It.IsInRange(1d, double.MaxValue, Range.Inclusive)).Verifiable();
+            mockTimer.Setup(x => x.Start()).Callback(() => timerRunning = true);
+            mockTimer.Setup(x => x.Stop()).Callback(() => timerRunning = false);
+            mockTimer.Setup(x => x.Dispose()).Callback(() => timerRunning = false);
+        }
+
+        [TestMethod]
+        public void Constructor_ThrowsOnNull()
+        {
+            Action op = () => new SonarQubeIssuesProvider(null, mockTracker.Object, mockTimerFactory.Object);
+            op.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("sonarQubeService");
+
+            op = () => new SonarQubeIssuesProvider(mockSqService.Object, null, mockTimerFactory.Object);
+            op.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("solutionBoundTracker");
+
+            op = () => new SonarQubeIssuesProvider(mockSqService.Object, mockTracker.Object, null);
+            op.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("timerFactory");
+        }
+
+        [TestMethod]
+        public void Constructor_TimerIsInitialized()
+        {
+            // Arrange
+            SetupSolutionBinding(true /* is bound */, false /* is connected */, null, null);
```

Comment:
That's completely optional, but I would replace the comments with the actual parameter names:
`SetupSolutionBinding(isBound:true, isConnected: false...)`

Reply:
Done.